src/main/java/ee/carlrobert/llm/client/watsonx/WatsonxClient.java

```diff
@@ -0,0 +1,163 @@
+package ee.carlrobert.llm.client.watsonx;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import ee.carlrobert.llm.PropertiesLoader;
+import ee.carlrobert.llm.client.DeserializationUtil;
+import ee.carlrobert.llm.client.openai.completion.ErrorDetails;
+import ee.carlrobert.llm.client.watsonx.completion.WatsonxCompletionRequest;
+import ee.carlrobert.llm.client.watsonx.completion.WatsonxCompletionResponse;
+import ee.carlrobert.llm.client.watsonx.completion.WatsonxCompletionResponseError;
+import ee.carlrobert.llm.completion.CompletionEventListener;
+import ee.carlrobert.llm.completion.CompletionEventSourceListener;
+import okhttp3.*;
+import okhttp3.sse.EventSource;
+import okhttp3.sse.EventSources;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static ee.carlrobert.llm.client.DeserializationUtil.OBJECT_MAPPER;
+
+public class WatsonxClient {
+
+        private static final MediaType APPLICATION_JSON = MediaType.parse("application/json");
+        private final OkHttpClient httpClient;
+        private final String host;
+        private final String apiVersion;
+        private final WatsonxAuthenticator authenticator;
+
+        private WatsonxClient(Builder builder, OkHttpClient.Builder httpClientBuilder) {
+            this.httpClient = httpClientBuilder.build();
+            this.apiVersion = builder.apiVersion;
+            this.host = builder.host;
+            if (builder.isOnPrem) {
+                if (builder.isZenApiKey)
+                    this.authenticator = new WatsonxAuthenticator(builder.username, builder.apiKey);
+                else
+                    this.authenticator = new WatsonxAuthenticator(builder.username, builder.apiKey, builder.host);
+            } else {
+                this.authenticator = new WatsonxAuthenticator(builder.apiKey);
+            }
+        }
+
+       public EventSource getCompletionAsync(
```

Comment:
Can we write some integration tests for these streaming methods? It will make refactoring easier later on. You can use other int. tests as an example.